zfmt/ast.go

```diff
@@ -636,11 +637,31 @@ func (c *canon) over(o *ast.Over) {
 		c.write(" with ")
 		c.defs(o.Locals, ", ")
 	}
-	if o.Scope != nil {
+	if o.Body != nil {
 		c.write(" => (")
 		c.open()
 		c.head = true
-		c.proc(o.Scope)
+		c.proc(o.Body)
+		c.close()
+		c.ret()
+		c.flush()
+		c.write(")")
+	}
+}
+
+func (c *canon) scope(s *ast.Scope, parens bool) {
+	if parens {
+		c.open("(")
+		c.ret()
+	}
+	for _, d := range s.Decls {
+		c.decl(d)
+		c.ret()
+	}
+	//XXX functions?
+	c.flush()
+	c.proc(s.Body)
+	if parens {
```

Comment:
Something's amiss here.

```console
$ go run ./cmd/zc -C 'a | (const b = 1 c)'
search a(
  const b = 1

  | search c
)
```

Reply:
Yes, I saw that.  Rather than spend time on this now, how about we make an issue to fix the tab/ret logic in zfmt to use a stack and simplify things.  It's hard to use and when I first put it together it was a quick hack.

Reply:
I can futz with this on a separate branch.

Reply:
Sounds good to me.

Reply:
#4534